### PR TITLE
[Feature/354832: Do you have a 2024 NPWD accreditation number?]

### PR DESCRIPTION
**Pull request by shehzad-ismail8**

### CODE REVIEW

File: src/EPR.Accreditation.Facade/Controllers/AccreditationSiteMaterialController.cs [hunk 1]
```diff
@@ -206,5 +206,31 @@ public class AccreditationSiteMaterialController : ControllerBase
...
             return Ok();
         }
+
+        [HttpGet("HasNpwdAccreditationNumber")]
+        public async Task<IActionResult> GetHasNpwdAccreditationNumber(
+            Guid accreditationExternalId,
+            Guid materialExternalId)
+        {
+            var hasNpwdAccreditationNumber = await _accreditationMaterialService.GetHasNpwdAccreditationNumber(
+                accreditationExternalId,
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationSiteMaterialController.cs]: Can we stop using the "external" bit please? Also refer to accreditationId as just id

File: src/EPR.Accreditation.Facade/Services/AccreditationMaterialService.cs [hunk 2]
```diff
@@ -142,5 +141,41 @@ public class AccreditationMaterialService : IAccreditationMaterialService
                 materialId,
                 material);
         }
+
+        public async Task<bool?> GetHasNpwdAccreditationNumber(
+            Guid accreditationExternalId,
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/AccreditationMaterialService.cs]: Remove external from param names. Then rename accreditationId to id

File: src/EPR.Accreditation.Facade/Services/AccreditationService.cs [hunk 3]
```diff
@@ -275,26 +275,28 @@ private Common.Dtos.Accreditation CreateNewAccreditationDto(OperatorType operato
...
         {
             var accreditation = await _httpAccreditationService.GetAccreditation(accreditationExternalId);
 
-            return accreditation == null 
-                ? new HasOverseasAgentDto() 
-                : new HasOverseasAgentDto 
-                    { 
-                        HasOverseasAgent = accreditation.HasOverseasAgent
-                    };
+            return accreditation == null
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/AccreditationService.cs:271]: Um.... if accreditation is null surely it doesn't exist? Therefore why are you returning valid objects? Surely a not found is the currect response?

File: src/EPR.Accreditation.Facade/Controllers/AccreditationSiteMaterialController.cs (hunk 1 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Facade/Controllers/AccreditationSiteMaterialController.cs]: Changed

File: src/EPR.Accreditation.Facade/Services/AccreditationMaterialService.cs (hunk 2 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Facade/Services/AccreditationMaterialService.cs]: Changed

**Review by NikPeters-Atos — APPROVED** [review]

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**